### PR TITLE
Add conditional and hidden field support to schema form

### PR DESCRIPTION
**Pull request by jbalboni**
Related to [870](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/870) and #4544 

This adds the ability to do conditionally displayed fields on the schemaform. This is done by adding two options to uiSchema: `updateSchema` and `hideIf`. Those are expected to be functions and both get the current page data and overall form data. The former can return a new schema (with updated `enum` values, for example) and the latter can return true or false.

Examples of these can be seen on the new school selection page.

screen shot 2017-02-10 at 12 42 17 pm
screen shot 2017-02-10 at 12 42 06 pm

### CODE REVIEW

**Review by cvalarida — APPROVED** [review]

File: src/js/common/schemaform/reducers/index.js [hunk 1]
```diff
@@ -31,13 +41,43 @@ export default function createSchemaFormReducer(formConfig) {
...
     switch (action.type) {
       case SET_DATA: {
+        // on each data change, we need to do the following steps
+        // Recalculate any required fields, based on the new data
+        let schema = updateRequiredFields(state[action.page].schema, state[action.page].uiSchema, action.data, state);
+        // Update the schema with any fields that are now hidden because of the data change
+        schema = setHiddenFields(schema, state[action.page].uiSchema, action.data, state);
+        // Update the schema with any general updates based on the new data
+        schema = updateSchemaFromUiSchema(schema, state[action.page].uiSchema, action.data, state);
+        // Remove any data that's how hidden in the schema
```
Comment (annekainicUSDS) [inline src/js/common/schemaform/reducers/index.js]: Minor type: I think you meant `now` instead of `how`

**Review by annekainicUSDS — COMMENTED** [review]
Testing this locally and I can't seem to get the top-up option to ever appear, assuming this works for you in testing locally?

**Comment by jbalboni** [comment]
@annekainicUSDS It should be working now.

**Comment by jbalboni** [comment]
@annekainicUSDS I removed the conditional logic for the top-up option, so it always shows up now.

**Review by annekainicUSDS — APPROVED** [review]
LGTM!

**Pull request merged**